### PR TITLE
IK-115[Refactor] Migrated matcher class into Object-Oriented style using Matcher class

### PR DESCRIPTION
**Pull request by anrchen**
Matcher class has been created, all matching processes are now handled by Matcher class. Unit tests have been updated too.

### CODE REVIEW

**Comment by anrchen** [comment]
Refactor the Matcher class into DomainModel folder.

**Review by adrielFab — CHANGES_REQUESTED** [review]
Some of the conditions do not follow coding convention, mainly the spacing.

File: client/app/src/androidTest/java/com/example/mrides/MatchRouteLogicTest.java [hunk 1]
```diff
@@ -37,21 +37,21 @@
     @Test
     // Test validateDistance method can match two coordinates within a reasonable distance
     public void testValideDistance(){
```
Comment (arosser) [inline client/app/src/androidTest/java/com/example/mrides/MatchRouteLogicTest.java]: missing space

File: client/app/src/androidTest/java/com/example/mrides/MatchRouteLogicTest.java [hunk 2]
```diff
@@ -37,21 +37,21 @@
...
 
-        assertEquals(routeActivity.validateDistance(start1, start2),
-                routeActivity.validateDistance(end1, end2));
+        assertEquals(matcher.validateDistance(start1, start2),
+                matcher.validateDistance(end1, end2));
     }
 
     @Test
     // Test the side effect of matchRoute() method on ArrayList <User> userOnMapCatalog
     public void testMatchRoute(){
```
Comment (arosser) [inline client/app/src/androidTest/java/com/example/mrides/MatchRouteLogicTest.java]: missing space

**Review by arosser — COMMENTED** [review]
Add a space between conditions and curly brackets.
ex: (condition) {

**Pull request merged**